test/src/test/java/org/springframework/security/test/web/servlet/request/SecurityMockMvcRequestBuildersFormLoginTests.java

```diff
@@ -82,6 +95,32 @@ public void customWithUriVars() {
 		assertThat(request.getRequestURI()).isEqualTo("/uri-login/val1/val2");
 	}
 
+	/**
+	 * spring-restdocs uses postprocessors to do its trick. It will work only if these are merged together
+	 * with our request builders. (gh-7572)
+	 * @throws Exception
+	 */
+	@Test
+	public void postProcessorsAreMergedDuringMockMvcPerform() throws Exception {
+
+		RequestBuilder requestBuilder = formLogin()
+				.user("my-user")
+				.password("my-password")
+				.loginProcessingUrl("/my-path");
+
+		MockHttpServletRequestBuilder defaultRequestBuilder = MockMvcRequestBuilders.get("/");
+		defaultRequestBuilder.with(new MockPostProcessor());
```

Comment:
Looking at this test again, I think some of this is a little incidentally verified. For example, if I remove the line above this test still passes. That means that it isn't verifying that the postProcessor is merged. I'm thinking that it might be better to use `MockMvc` to perform the validation. Something like this:

```java
	@Test
	public void postProcessorsAreMergedDuringMockMvcPerform() throws Exception {
		RequestPostProcessor postProcessor = mock(RequestPostProcessor.class);
		when(postProcessor.postProcessRequest(any())).thenAnswer(i -> i.getArgument(0));
		MockMvc mockMvc = MockMvcBuilders.standaloneSetup(new Object())
				.defaultRequest(MockMvcRequestBuilders.get("/").with(postProcessor))
				.build();

		mockMvc.perform(formLogin());

		verify(postProcessor).postProcessRequest(any());
	}
	
```

Thoughts?

Reply:
Your approach seems to be much better, I will use this :)